plugins/BEdita/API/src/Controller/Model/SchemaController.php

```diff
@@ -26,14 +26,26 @@
  */
 class SchemaController extends AppController
 {
+    /**
+     * JSON Schema content type.
+     *
+     * @var string
+     */
+    const CONTENT_TYPE = 'application/schema+json';
+
     /**
      * {@inheritDoc}
      */
     public function initialize()
     {
         parent::initialize();
-        $this->components()->unload('JsonApi');
-        $this->viewBuilder()->setClassName('Json');
+        if ($this->components()->get('JsonApi')) {
+            $this->components()->unload('JsonApi');
+        }
+        $this->viewBuilder()->setClassName('BEdita\API\View\JsonSchemaView');
```

Comment:
Frankly I don't like having a custom view class just for the sake of this scope… It looks quite much a bit overkill to me.

I'd use a different and simpler approach, even though it is less reproducible:
```diff
--- plugins/BEdita/API/src/Controller/Model/SchemaController.php	(revision 128981534193332d8b07aec4cbec29da075e4483)
+++ plugins/BEdita/API/src/Controller/Model/SchemaController.php	(revision )
@@ -16,7 +16,6 @@
 use BEdita\API\Controller\AppController;
 use BEdita\Core\Model\Schema\JsonSchema;
 use Cake\Event\Event;
-use Cake\Routing\Router;
 
 /**
  * Controller for `/model/schema/{type}` endpoint.
@@ -39,13 +38,11 @@
     public function initialize()
     {
         parent::initialize();
+
         if ($this->components()->get('JsonApi')) {
             $this->components()->unload('JsonApi');
         }
-        $this->viewBuilder()->setClassName('BEdita\API\View\JsonSchemaView');
-        $this->response->type([
-            'jsonschema' => self::CONTENT_TYPE,
-        ]);
+        $this->viewBuilder()->setClassName('Json');
     }
 
     /**
@@ -64,7 +61,7 @@
      * Get JSON-SCHEMA of a type.
      *
      * @param string $typeName Name of an object type or of a resource type.
-     * @return void
+     * @return \Cake\Http\Response
      */
     public function jsonSchema($typeName)
     {
@@ -73,5 +70,7 @@
         $url = (string)$this->request->getUri();
         $this->set(JsonSchema::generate($typeName, $url));
         $this->set('_serialize', true);
+
+        return $this->render()->withType(static::CONTENT_TYPE);
     }
 }
```

Reply:
That's right, I quickly tried similar solutions but didn't find this one.